app/src/utils/rules/misc.js

```diff
@@ -231,18 +231,28 @@ export const fixRegexStr = (regexStr) => {
 export function runMinorFixesOnRule(dispatch, rule) {
   const rulePairs = rule.pairs.map((pair) => {
     let fixedPair = pair;
+
     // fix regex
     if (pair.source.operator === GLOBAL_CONSTANTS.RULE_OPERATORS.MATCHES) {
+      let fixedRegex = pair.source.value;
       if (!isValidRegex(pair.source.value)) {
-        fixedPair = {
-          ...pair,
-          source: {
-            ...pair.source,
-            value: fixRegexStr(pair.source.value),
-          },
-        };
+        fixedRegex = fixRegexStr(pair.source.value);
+      }
+
+      // if replace rule, add non-capturing group to all capturing groups
+      if (rule.ruleType === GLOBAL_CONSTANTS.RULE_TYPES.REPLACE) {
+        fixedRegex = fixedRegex.replaceAll(/\((?!\?:)/g, "(?:");
```

Comment:
Fails for conditions which contains lookarounds as they also contains brackets
<img width="292" alt="CleanShot 2024-07-08 at 18 32 25@2x" src="https://github.com/requestly/requestly/assets/16779465/21252b34-f4a0-427e-9b25-34d239363e87">